src/dispatching/dispatcher.rs

```diff
@@ -205,13 +228,20 @@ where
             }
         }
 
-        // TODO: wait for executing handlers?
+        for (_chat_id, worker) in self.workers.drain() {
+            drop(worker.tx);
+            worker.handle.await.expect("Unable to wait for a worker");
+        }
+        if let Some(worker) = self.default_worker.take() {
+            drop(worker.tx);
+            worker.handle.await.expect("Unable to wait for a default handler");
+        }
```

Comment:
```suggestion
        self.workers.drain()
            .map(|(_chat_id, worker)| worker.handle)
            .chain(self.default_worker.take().map(|worker| worker.handle))        
            .collect::<FuturesUnordered<_>>()
            .for_each(|()| ())
            .await;
```

The reason to do it this way is that we first drop all `.tx` (signalling to the workers that it's time to stop) and then wait for all the handlers at the same time (in your current implementation you drop&wait one-by-one).